client/src/ProductDetails.jsx

```diff
@@ -47,12 +50,12 @@ export default function ProductDetails() {
         });
     }
 
-    return (
+    return authenticated ? (
```

Comment:
I am not happy with the way I am checking if the user is authenticated multiple times 
I wrote about this in the commit message.